webapp/packages/plugin-connection-template/src/ConnectionDialog/useConnectionDialog.ts

```diff
@@ -0,0 +1,208 @@
+
+          if (connection.requiredAuth) {
+            const state = await this.authProviderService.requireProvider(connection.requiredAuth);
+
+            if (!state) {
+              this.setStep(ConnectionStep.ConnectionTemplateSelect);
+              await this.connectionInfoResource.deleteConnection(createConnectionParam(connection));
+              return;
+            }
+          }
+
+          try {
+            await this.connectionInfoResource.init(this.getConfig(connection.projectId, connection.id));
+            this.notificationService.logSuccess({ title: 'plugin_connection_template_connect_success', message: connection.name });
+            this.onConnect?.();
+          } catch (exception: any) {
+            this.connectException = exception;
+            await this.connectionInfoResource.deleteConnection(createConnectionParam(connection));
+          }
+        } catch (exception: any) {
+          this.connectException = exception;
+        } finally {
+          this.processing = false;
+        }
+      },
+      getConfig(projectId: string, connectionId: string) {
+        const config: ConnectionInitConfig = {
+          projectId,
+          connectionId,
+        };
+
+        if (this.authModelId) {
+          if (Object.keys(this.config.credentials).length > 0) {
+            config.credentials = this.config.credentials;
+          }
+
+          config.saveCredentials = this.config.saveCredentials;
+        }
+
+        if (this.config.networkHandlersConfig.length > 0) {
+          config.networkCredentials = this.config.networkHandlersConfig;
+        }
+
+        return config;
+      },
+      async selectTemplate(templateId: string) {
+        this.template = this.templateConnectionsService.projectTemplates.find(template => template.id === templateId)!;
+        this.driver = await this.dbDriverResource.load(this.template.driverId);
+        this.config = getDefaultConfig();
+
+        if (this.template.authNeeded) {
+          const property = this.template.authProperties?.find(property => property.id === USER_NAME_PROPERTY_ID);
+
+          if (property?.value) {
+            this.config.credentials[USER_NAME_PROPERTY_ID] = property.value;
+          }
+        }
+
+        for (const id of this.networkHandlers) {
+          const handler = this.template.networkHandlersConfig?.find(handler => handler.id === id);
+
+          if (handler && (handler.userName || handler.authType !== NetworkHandlerAuthType.Password)) {
+            this.config.networkHandlersConfig.push({
+              id: handler.id,
+              authType: handler.authType,
+              userName: handler.userName,
+              password: handler.password,
+              savePassword: handler.savePassword,
+            });
+          }
+        }
+
+        this.step = ConnectionStep.Connection;
```

Comment:
please use `this.setStep(ConnectionStep.Connection)` also you probably want to call it right after setting template and default config (synchronously)